src/common/njord.ts

```diff
@@ -531,6 +538,38 @@ const canAward = async ({
   }
 };
 
+export const awardSquad = async (
+  props: AwardInput,
+  ctx: AuthContext,
+): Promise<TransactionCreated> => {
+  const sourceId = props.flags?.sourceId;
+  if (!sourceId) {
+    throw new ForbiddenError('You can not award this Squad');
+  }
+  try {
+    await ctx.con.manager.getRepository(Source).findOneOrFail({
+      select: ['id'],
+      where: {
+        id: sourceId,
+      },
+    });
+    return awardUser(props, ctx);
+  } catch (error) {
+    logger.error(
+      {
+        context: 'award',
+        err: error,
+        props,
+        userId: ctx.userId,
+        transactionId: newTransactionId,
```

Comment:
you are missing transaction here, variable is not defined

Reply:
I think you can just call `awardUser` as well, you don't need to catch it since it will catch inside of itself

Reply:
Also not to catch issues on the squad query?